src/fd_integration/partial_state.h

```diff
@@ -15,8 +15,10 @@ class StateInterface;
 class PR2State : public StateInterface {
     std::vector<int> vars; // values for vars
     std::vector< std::pair<int,int> > * _varvals = NULL; // varval pairs for partial states
-    // void _allocate();
-    // void _deallocate();
+    void _allocate(int size) {
+        vars.reserve(size);
+    }
+    void _deallocate() {}
```

Comment:
Shouldn't the memory be released?

Reply:
I may be wrong, but I believe the C++ vector should automatically deallocate everything when it goes out of scope. Looking it up reserve doesn't seem to require a free method